### PR TITLE
feat: add defered_ephemeral property

### PR DESCRIPTION
**Pull request by GeomKid**
## About

This pull request is about add defered_ephemeral property, which does .

## Checklist

- [ ] The ``pre-commit`` code linter has been run over all edited files to ensure the code is linted.
- [ ] I've ensured the change(s) work on `3.8.6` and higher.

I've made this pull request: (check all that apply)
  - [x] For the documentation
  - [x] To add a new feature
  - [ ] As a general enhancement
  - [ ] As a refactor of the library/the library's code
  - [ ] To fix an existing bug
  - [ ] To resolve #ISSUENUMBER

This is:
  - [ ] A breaking change

### CODE REVIEW

File: interactions/client/context.py [hunk 1]
```diff
@@ -742,7 +751,7 @@ async def defer(
 
             self.deferred = True
             _ephemeral: int = MessageFlags.EPHEMERAL.value if bool(ephemeral) else 0
-
+            self.deferred_ephemeral = bool(_ephemeral)
```
Comment (EdVraz) [inline interactions/client/context.py:754]: This is not needed

File: interactions/client/context.py [hunk 2]
```diff
@@ -60,6 +60,7 @@ class _Context(ClientSerializerMixin):
     responded: bool = field(default=False)
     deferred: bool = field(default=False)
     app_permissions: Permissions = field(converter=convert_int(Permissions), default=None)
+    deferred_ephemeral: bool = field(default=False)
```
Comment (sillyElvy) [inline interactions/client/context.py]: Please make it a property that uses `self.message.flags`

**Review by sillyElvy — CHANGES_REQUESTED** [review]

File: interactions/client/context.py [hunk 3]
```diff
@@ -384,6 +391,7 @@ class CommandContext(_Context):
     :ivar bool responded: Whether an original response was made or not.
     :ivar bool deferred: Whether the response was deferred or not.
     :ivar Optional[Locale] locale: The selected language of the user invoking the interaction.
+    :ivar bool deferred_ephemeral: Whether the response was deferred and ephemeral.
```
Comment (sillyElvy) [inline interactions/client/context.py]:
```suggestion
```

File: interactions/client/context.py [hunk 4]
```diff
@@ -63,6 +63,13 @@ class _Context(ClientSerializerMixin):
     locale: Optional[Locale] = field(converter=Locale, default=None)
     guild_locale: Optional[Locale] = field(converter=Locale, default=None)
 
+    @property
+    def deferred_ephemeral(self) -> bool:
```
Comment (sillyElvy) [inline interactions/client/context.py:67]:
```suggestion
    def deferred_ephemeral(self) -> bool:
    """Returns whether the current interaction was deferred ephemerally."""
```

File: interactions/client/context.py [hunk 5]
```diff
@@ -63,6 +63,13 @@ class _Context(ClientSerializerMixin):
...
+        return bool(
```
Comment (sillyElvy) [inline interactions/client/context.py:73]:
```suggestion
        return (
```

No unnecessary casts to bool :)

**Review by sillyElvy — CHANGES_REQUESTED** [review]

**Comment by sillyElvy** [comment]
Also please test it

File: interactions/client/context.py (hunk 5 above)
Comment (GeomKid) [reply interactions/client/context.py:73]: I tried without the bool it returns the enum
```
b4 adding bool
INFO:root:MessageFlags.LOADING
after adding bool
INFO:root:True
```
My logs when b4 I add bool and after add bool

Comment (sillyElvy) [reply interactions/client/context.py:73]: That's interesting, bool can stay then

(However the Enum would still be true if its not 0)

File: interactions/client/context.py [hunk 6]
```diff
@@ -63,6 +63,16 @@ class _Context(ClientSerializerMixin):
     locale: Optional[Locale] = field(converter=Locale, default=None)
     guild_locale: Optional[Locale] = field(converter=Locale, default=None)
 
+    @property
+    def deferred_ephemeral(self) -> bool:
+        """
```
Comment (sillyElvy) [inline interactions/client/context.py:68]:
```suggestion
        """
        .. versionadded:: 4.4.0

```

**Review by sillyElvy — CHANGES_REQUESTED** [review]
We might as well make defer return the message object now

File: interactions/client/context.py [hunk 7]
```diff
@@ -488,25 +501,34 @@ async def edit(
 
         return msg if msg is not None else Message(**payload, _client=self._client)
 
-    async def defer(self, ephemeral: Optional[bool] = False) -> None:
+    async def defer(self, ephemeral: Optional[bool] = False) -> Message:
         """
```
Comment (sillyElvy) [inline interactions/client/context.py:504]:
```suggestion
        """
        .. versionchanged::; 4.4.0
            Now returns the created message object
            
```

File: interactions/client/context.py [hunk 8]
```diff
@@ -730,19 +752,20 @@ async def send(self, content: Optional[str] = MISSING, **kwargs) -> Message:
 
     async def defer(
         self, ephemeral: Optional[bool] = False, edit_origin: Optional[bool] = False
-    ) -> None:
+    ) -> Message:
         """
```
Comment (sillyElvy) [inline interactions/client/context.py:758]:
```suggestion
        """
        .. versionchanged::; 4.4.0
            Now returns the created message object
            
```

File: interactions/client/context.py [hunk 9]
```diff
@@ -488,25 +501,34 @@ async def edit(
...
         return msg if msg is not None else Message(**payload, _client=self._client)
 
-    async def defer(self, ephemeral: Optional[bool] = False) -> None:
+    async def defer(self, ephemeral: Optional[bool] = False) -> Message:
         """
         This "defers" an interaction response, allowing up
         to a 15-minute delay between invocation and responding.
 
         :param Optional[bool] ephemeral: Whether the deferred state is hidden or not.
+        :return: The defering message
```
Comment (sillyElvy) [inline interactions/client/context.py]:
```suggestion
        :return: The deferred message
```

File: interactions/client/context.py [hunk 10]
```diff
@@ -730,19 +752,20 @@ async def send(self, content: Optional[str] = MISSING, **kwargs) -> Message:
...
         self, ephemeral: Optional[bool] = False, edit_origin: Optional[bool] = False
-    ) -> None:
+    ) -> Message:
         """
         This "defers" a component response, allowing up
         to a 15-minute delay between invocation and responding.
 
         :param Optional[bool] ephemeral: Whether the deferred state is hidden or not.
         :param Optional[bool] edit_origin: Whether you want to edit the original message or send a followup message
+        :return: The defering message
```
Comment (sillyElvy) [inline interactions/client/context.py]:
```suggestion
        :return: The deferred message
```

File: interactions/client/context.py [hunk 11]
```diff
@@ -63,6 +63,19 @@ class _Context(ClientSerializerMixin):
     locale: Optional[Locale] = field(converter=Locale, default=None)
     guild_locale: Optional[Locale] = field(converter=Locale, default=None)
 
+    @property
+    def deferred_ephemeral(self) -> bool:
+        """
+        .. versionadded:: 4.4.0
+
+        .. versionadded:: 4.4.0
```
Comment (sillyElvy) [inline interactions/client/context.py]:
```suggestion
```

**Review by sillyElvy — CHANGES_REQUESTED** [review]

File: interactions/client/context.py [hunk 12]
```diff
@@ -488,25 +500,36 @@ async def edit(
 
         return msg if msg is not None else Message(**payload, _client=self._client)
 
-    async def defer(self, ephemeral: Optional[bool] = False) -> None:
+    async def defer(self, ephemeral: Optional[bool] = False) -> Message:
         """
+        .. versionchanged::; 4.4.0
+            Now returns the created message object
```
Comment (sillyElvy) [inline interactions/client/context.py:506]:
```suggestion
            Now returns the created message object
            
```

File: interactions/client/context.py [hunk 13]
```diff
@@ -730,19 +753,22 @@ async def send(self, content: Optional[str] = MISSING, **kwargs) -> Message:
 
     async def defer(
         self, ephemeral: Optional[bool] = False, edit_origin: Optional[bool] = False
-    ) -> None:
+    ) -> Message:
         """
+        .. versionchanged::; 4.4.0
+            Now returns the created message object
```
Comment (sillyElvy) [inline interactions/client/context.py:760]:
```suggestion
            Now returns the created message object
            
```

**Review by sillyElvy — CHANGES_REQUESTED** [review]

File: interactions/client/context.py [hunk 14]
```diff
@@ -488,25 +500,37 @@ async def edit(
 
         return msg if msg is not None else Message(**payload, _client=self._client)
 
-    async def defer(self, ephemeral: Optional[bool] = False) -> None:
+    async def defer(self, ephemeral: Optional[bool] = False) -> Message:
         """
+        .. versionchanged::; 4.4.0
```
Comment (sillyElvy) [inline interactions/client/context.py]:
```suggestion
        .. versionchanged:: 4.4.0
```

File: interactions/client/context.py [hunk 15]
```diff
@@ -730,19 +754,23 @@ async def send(self, content: Optional[str] = MISSING, **kwargs) -> Message:
 
     async def defer(
         self, ephemeral: Optional[bool] = False, edit_origin: Optional[bool] = False
-    ) -> None:
+    ) -> Message:
         """
+        .. versionchanged::; 4.4.0
```
Comment (sillyElvy) [inline interactions/client/context.py]:
```suggestion
        .. versionchanged:: 4.4.0
```

**Review by sillyElvy — CHANGES_REQUESTED** [review]

**Review by sillyElvy — APPROVED** [review]
i think its fine now

**Pull request merged**